tests/test_getPlotDatasetTypeNames.py

```diff
@@ -0,0 +1,80 @@
+# This file is part of analysis_tools.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+from __future__ import annotations
+
+import os
+from unittest import TestCase, main
+
+import lsst.utils.tests
+from lsst.analysis.tools.analysisPlots.skyObject import SkyObjectHistPlot
+from lsst.analysis.tools.tasks.objectTableTractAnalysis import ObjectTableTractAnalysisConfig
+from lsst.analysis.tools.tasks.reconstructor import getPlotDatasetTypeNames
+from lsst.daf.butler import Butler, DatasetType
+from lsst.daf.butler.tests.utils import makeTestTempDir, removeTestTempDir
+from lsst.pex.config import Config
+
+TESTDIR = os.path.abspath(os.path.dirname(__file__))
+
+
+class TestGetPlotDatasetTypeNames(TestCase):
+    """Test plot dataset type names are correctly queried by the butler."""
+
+    def setUp(self):
+        self.root = makeTestTempDir(TESTDIR)
+        Butler.makeRepo(self.root)
+        self.butler = Butler(self.root, run="test_run")
+        # No dimensions in dataset type so we don't have to worry about
+        # inserting dimension data or defining data IDs.
+        self.datasetTypePlot = DatasetType(
+            "plotType_config", dimensions=(), storageClass="Config", universe=self.butler.registry.dimensions
+        )
+        self.butler.registry.registerDatasetType(self.datasetTypePlot)
+        self.datasetTypeOther = DatasetType(
+            "otherType_config", dimensions=(), storageClass="Config", universe=self.butler.registry.dimensions
+        )
+        self.butler.registry.registerDatasetType(self.datasetTypeOther)
+
+    def tearDown(self):
+        removeTestTempDir(self.root)
+
+    def testGetPlotNames(self) -> None:
```

Comment:
This is a good test, but you also should have a test of the `label` if-branch of the code.